homeassistant/components/philips_js/media_player.py

```diff
@@ -146,38 +145,18 @@ def state(self):
     @property
     def source(self):
         """Return the current input source."""
-        if self.media_content_type == MEDIA_TYPE_CHANNEL:
-            name = self._channels.get(self._tv.channel_id)
-            prefix = PREFIX_CHANNEL
-        else:
-            name = self._sources.get(self._tv.source_id)
-            prefix = PREFIX_SOURCE
-
-        if name is None:
-            return None
-        return prefix + PREFIX_SEPARATOR + name
+        return self._sources.get(self._tv.source_id)
 
     @property
     def source_list(self):
         """List of available input sources."""
-        complete = []
-        for source in self._sources.values():
-            complete.append(PREFIX_SOURCE + PREFIX_SEPARATOR + source)
-        for channel in self._channels.values():
-            complete.append(PREFIX_CHANNEL + PREFIX_SEPARATOR + channel)
-        return complete
+        return list(self._sources.values())
 
     def select_source(self, source):
         """Set the input source."""
-        data = source.split(PREFIX_SEPARATOR, 1)
-        if data[0] == PREFIX_SOURCE:
-            source_id = _inverted(self._sources).get(data[1])
-            if source_id:
-                self._tv.setSource(source_id)
-        elif data[0] == PREFIX_CHANNEL:
-            channel_id = _inverted(self._channels).get(data[1])
-            if channel_id:
-                self._tv.setChannel(channel_id)
+        source_id = _inverted(self._sources).get(source)
+        if source_id:
+            self._tv.setSource(source_id)
```

Comment:
Could we keep the old logic here for a while so we don't break existing scripts ?

Reply:
I suppose, but I really think it's more confusing than anything. I feel ripping the band-aid make more sense.

Reply:
new users won't know they can pass it in, because it's not shown in the source list. If we can and it's not in the way, let's not break things.

Reply:
Oh wait you just the specific play media case. Yes that we can keep around for a while.